v2/json2/server.go

```diff
@@ -128,12 +128,18 @@ func (c *CodecRequest) ReadRequest(args interface{}) error {
 	if c.err == nil && c.request.Params != nil {
 		// Note: if c.request.Params is nil it's not an error, it's an optional member.
 		// JSON params structured object. Unmarshal to the args object.
-		err := json.Unmarshal(*c.request.Params, args)
-		if err != nil {
-			c.err = &Error{
-				Code:    E_INVALID_REQ,
-				Message: err.Error(),
-				Data:    c.request.Params,
+		if err := json.Unmarshal(*c.request.Params, args); err != nil {
```

Comment:
This looks OK, but can you embed the RFC in the docs to make this fallback more clear/explain the rationale behind it? Will merge once that's done.